specreduce/extract.py

```diff
@@ -3,216 +3,162 @@
-            uncertainty=StdDevUncertainty(fluxerr),
-            mask=mask
-        )
-        skyspec = Spectrum1D(
-            spectral_axis=np.arange(len(onedspec)) * u.pixel,
-            flux=skysubflux * img_unit,
-            mask=mask
-        )
-
-        return spec, skyspec
-
-    def _extract_from_box(self, image, wave_index, low_end, high_end, extracted_result):
-
-        # compute nearest integer endpoints defining an internal interval,
-        # and fractional pixel areas that remain outside this interval.
-        # (taken from the HST STIS pipeline code:
-        # https://github.com/spacetelescope/hstcal/blob/master/pkg/stis/calstis/cs6/x1dspec.c)
-        #
-        # This assumes that the pixel coordinates represent the center of the pixel.
-        # E.g. pixel at y=15.0 covers the image from y=14.5 to y=15.5
-
-        # nearest integer endpoints
-        j1 = self._find_nearest_int(low_end)
-        j2 = self._find_nearest_int(high_end)
-
-        # fractional pixel areas at the end points
-        s1 = 0.5 - (low_end - j1)
-        s2 = 0.5 + high_end - j2
-
-        # add up the total flux around the trace_line
-        extracted_result[wave_index] = np.nansum(image[j1 + 1:j2, wave_index])
-        extracted_result[wave_index] += np.nansum(image[j1, wave_index]) * s1
-        extracted_result[wave_index] += np.nansum(image[j2, wave_index]) * s2
-
-    def _find_nearest_int(self, end_point):
-        if (end_point % 1) < 0.5:
-            return int(end_point)
-        else:
-            return int(end_point + 1)
-
-    def get_checkplot(self):
-        trace_line = self.last_trace.line
-
-        fig = plt.figure()
-        plt.imshow(self.last_img, origin='lower', aspect='auto', cmap=plt.cm.Greys_r)
-        plt.clim(np.percentile(self.last_img, (5, 98)))
-
-        plt.plot(np.arange(len(trace_line)), trace_line, c='C0')
-        plt.fill_between(
-            np.arange(len(trace_line)),
-            trace_line + self.apwidth,
-            trace_line - self.apwidth,
-            color='C0',
-            alpha=0.5
-        )
-        plt.fill_between(
-            np.arange(len(trace_line)),
-            trace_line + self.apwidth + self.skysep,
-            trace_line + self.apwidth + self.skysep + self.skywidth,
-            color='C1',
-            alpha=0.5
-        )
-        plt.fill_between(
-            np.arange(len(trace_line)),
-            trace_line - self.apwidth - self.skysep,
-            trace_line - self.apwidth - self.skysep - self.skywidth,
-            color='C1',
-            alpha=0.5
-        )
-        plt.ylim(
-            np.min(
-                trace_line - (self.apwidth + self.skysep + self.skywidth) * 2
-            ),
-            np.max(
-                trace_line + (self.apwidth + self.skysep + self.skywidth) * 2
-            )
-        )
-
-        return fig
+    # TODO: what is a reasonable default?
+    # TODO: int or float?
+    width: int = 5
```

Comment:
the `_ap_weight_image` function says it takes `width` as a float, so should we make it a float here as well?  Or do we want to leave it as an integer (as was the case for the previous implementation)?

Reply:
I think it should be a float. The code works in the same way with floats or integers, and float is more general.

Reply:
agreed.